src/newcharacter.cpp

```diff
@@ -2888,11 +2888,17 @@ std::vector<trait_id> Character::get_base_traits() const
     return std::vector<trait_id>( my_traits.begin(), my_traits.end() );
 }
 
-std::vector<trait_id> Character::get_mutations( bool include_hidden ) const
+std::vector<trait_id> Character::get_mutations( bool include_hidden, mutation_filter filter ) const
 {
     std::vector<trait_id> result;
     for( const std::pair<const trait_id, trait_data> &t : my_mutations ) {
         if( include_hidden || t.first.obj().player_display ) {
+            if( filter == mutation_filter::anger_relations && t.first.obj().anger_relations.empty() ) {
+                break;
+            }
+            if( filter == mutation_filter::ignored_by && t.first.obj().ignored_by.empty() ) {
+                break;
```

Comment:
Should be continue if I'm not mistaken.
```suggestion
                continue;
            }
            if( filter == mutation_filter::ignored_by && t.first.obj().ignored_by.empty() ) {
                continue;
```

Reply:
Thanks!